### DISCUSSION TITLE
Can you use GitHub Action account to approve PRs?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#27090] [Actions]
<p>I have a repository and want to enforce a branch system by having the master branch protected and requiring at least 1 approving review for the Pull request to be mergeable.</p>
<p>Right now am I the only person contributing to the project and I (ab)use my admin privileges to bypass this approval-requirement since I can’t approve my own PRs (Can I tho? Like is there a way?)<br>
However, I don’t like this as it feels a bit cheaty and I want to prevent others with those rights to easily bypass this since the project is part of an org.</p>
<p>My question now is, if I can use the GitHub Actions account to approve the changes, so that the PR would become mergeable.<br>
If that is doable would I like to use a System with either assigning a label (Something like “Action: Auto-Approve”) or writing a comment like “pls approve” to make the action then approve the PR.</p>
<p>I hope that there is a solution for this. Thanks in advance for any solutions.</p>

**Comment by Simran-B** [comment, marked as answer]
<aside class="quote no-group" data-username="Andre601" data-post="1" data-topic="170834">
<div class="title">
<div class="quote-controls"></div>
 Andre601:</div>
<blockquote>
<p>I can’t approve my own PRs (Can I tho? Like is there a way?)</p>
</blockquote>
</aside>
<p>I don’t think so. At least the GitHub web interface does not let you do that. I’m not sure if the API might allow it.</p>
<blockquote>
<p>I want to prevent others with [admin privileges] to easily bypass this [approval-requirement]</p>
</blockquote>
<p>As I understand, you still want to be able to bypass the approval process for yourself. Are you an administrator too or the owner of the repo? I suppose that the owner can still bypass the approval even if they are mandatory for admins (“Include administrators” at the bottom):</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/3/13c9613b88d9ff206510d209d16f500bb59180ab.png" data-download-href="/uploads/short-url/2P2wCIA4fS4Gpf51RWv5gndAvEL.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7819991/181101637-4b6ee513-89d4-4c9c-800b-5e0d05bb8e0d.png" alt="image" data-base62-sha1="2P2wCIA4fS4Gpf51RWv5gndAvEL" width="675" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/3/13c9613b88d9ff206510d209d16f500bb59180ab_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">922×682 43.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><p></p>
<p>If that doesn’t work for you, then yes, you can let a workflow approve your pull requests, e.g. using the <code>github-script</code> action:</p>
<pre><code class="lang-yaml">name: PR Approval

on:
  pull_request:
    branches:
      - master
  types: [opened] # probably the only one you need

jobs:
  build:
    name: Auto-approve
    runs-on: ubuntu-latest
    if: github.event.pull_request.user.login == 'andre601'
    steps:
      - uses: actions/github-script@v4
        with:
          script: |
            const result = await github.pulls.createReview({
              owner: context.repo.owner,
              repo: context.repo.repo,
              pull_number: context.issue.number,
              event: "APPROVE"
            })
            console.log(result.status)
</code></pre>
<p><img src="https://user-images.githubusercontent.com/7819991/181101638-e5c45c00-2455-4250-bc7c-31fd467a27c1.png" alt="image" data-base62-sha1="lG2fGgG8FumXer91dQHsJn4w0Sm" width="513" height="73"></p>
<p>You may need to restrict who can approve PRs though. I saw spam bots randomly approving PRs in public repos, for instance.</p>

**Reply by andretai** [reply]
Hi @Simran-B , just to confirm, we can configure a bot with approving privileges yes? Meaning when the rule is to require pull request with at least **1 approving review**, the bot qualifies as one who gives that approving review.

**Comment by Andre601** [comment]
<p>I used a action for auto-approving Prs with an if statement and works fine. Thanks for the info and clarification tho.</p>
